cmd/kubeadm/app/features/features.go

```diff
@@ -27,37 +27,22 @@ import (
 )
 
 const (
-	// HighAvailability is alpha in v1.9 - deprecated in v1.12 (TODO remove in v1.13)
-	HighAvailability = "HighAvailability"
 
 	// CoreDNS is GA in v1.11
 	CoreDNS = "CoreDNS"
 
-	// SelfHosting is alpha in v1.8 and v1.9 - deprecated in v1.12 (TODO remove in v1.13)
+	// SelfHosting is alpha in v1.8 and v1.9 - deprecated in v1.12 - removed from workflow in v1.13
 	SelfHosting = "SelfHosting"
 
-	// StoreCertsInSecrets is alpha in v1.8 and v1.9 - deprecated in v1.12 (TODO remove in v1.13)
-	StoreCertsInSecrets = "StoreCertsInSecrets"
-
 	// DynamicKubeletConfig is beta in v1.11
 	DynamicKubeletConfig = "DynamicKubeletConfig"
 
 	// Auditing is beta in 1.8
 	Auditing = "Auditing"
 )
 
-var selfHostingDeprecationMessage = "featureGates:SelfHosting has been removed in v1.12"
-
-var storeCertsInSecretsDeprecationMessage = "featureGates:StoreCertsInSecrets has been removed in v1.12"
-
-var highAvailabilityMessage = "featureGates:HighAvailability has been removed in v1.12\n" +
-	"\tThis feature has been replaced by the kubeadm join --control-plane workflow."
-
 // InitFeatureGates are the default feature gates for the init command
 var InitFeatureGates = FeatureList{
-	SelfHosting:          {FeatureSpec: utilfeature.FeatureSpec{Default: false, PreRelease: utilfeature.Deprecated}, HiddenInHelpText: true, DeprecationMessage: selfHostingDeprecationMessage},
-	StoreCertsInSecrets:  {FeatureSpec: utilfeature.FeatureSpec{Default: false, PreRelease: utilfeature.Deprecated}, HiddenInHelpText: true, DeprecationMessage: storeCertsInSecretsDeprecationMessage},
-	HighAvailability:     {FeatureSpec: utilfeature.FeatureSpec{Default: false, PreRelease: utilfeature.Deprecated}, HiddenInHelpText: true, DeprecationMessage: highAvailabilityMessage},
```

Comment:
this PR is not only about self-hosting it seems.
probably should have been broken down in further steps, first being "removing the f-gates for x, y, z".

Reply:
@neolit123 and @fabriziopandini mostly this does just remove self-hosting feature gates and the code that calls the feature gates. The only thing on top of that is that it removed the selfhosting.go file from the phases folder as it is not a phase anymore with the feature gate removed.

I was under the impression the `StoreCertsInSecrets` and `HighAvailability` where both features related to selfhosting and if you removed self hosting you removed both of them. If I am wrong I am happy to correct. I too felt like the PR is way to large, but minus a few small update that will be coming later today I felt like this was a good breakup of the work.

Reply:
I'm fine in removing all the three feature gates in one PR